tools/cbmc/proofs/DNS/DNSgetHostByName/DNSgetHostByName_harness.c

```diff
@@ -0,0 +1,47 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_DNS.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_Sockets.h"
+
+/* This assumes the length of pcHostName is bounded by MAX_HOSTNAME_LEN and the size of UDPPayloadBuffer is bounded by 
+MAX_REQ_SIZE. */ 
+
+void *safeMalloc(size_t xWantedSize) {
+	if(xWantedSize == 0) {
+		return NULL;
+	}
+	uint8_t byte;
+	return byte ? malloc(xWantedSize) : NULL;
+}
+
+/* Abstraction of FreeRTOS_GetUDPPayloadBuffer. This should be checked later. For now we are allocating a fixed sized memory of size MAX_REQ_SIZE. */
+void * FreeRTOS_GetUDPPayloadBuffer(size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks ) {
+	void *pvReturn = safeMalloc(MAX_REQ_SIZE); 
+	return pvReturn;
+}
+
+/* Abstraction of FreeRTOS_socket. This abstraction allocates a memory of size Socket_t. */
+Socket_t FreeRTOS_socket( BaseType_t xDomain, BaseType_t xType, BaseType_t xProtocol ) {
+	Socket_t xSocket = safeMalloc(sizeof(Socket_t)); /* Replaced malloc by safeMalloc */
+	return xSocket;
+}
+
+/* This function only uses the return value of prvParseDNSReply. Hence it returns an unconstrained uint32 value */
+uint32_t prvParseDNSReply( uint8_t *pucUDPPayloadBuffer, size_t xBufferLength, TickType_t xIdentifier) { }
```

Comment:
There seems to be a lot of duplication among the DNS harness files.  For the sake of future maintainers, can we strive for DRY, and refactor so this code is shared?

Reply:
@dan4thewin May I request that I do the deduplication work myself, later in a separate PR?

There has certainly been a lot of duplicate stubs in the proofs that we've been adding this summer. I was planning to re-review them myself once everything has settled down a few weeks from now, and decide on the most sensible way to unify the stubs.

In AWS C-Common (another project that we've been contributing proofs to), there is a [proof helpers directory](https://github.com/awslabs/aws-c-common/tree/master/.cbmc-batch/include/proof_helpers) containing code that is common to all proofs. It's certainly a good idea to have a similar thing in a:FR, but I'd like to get an overview of the commonality in the current and incoming proofs first.

Please let me know if that's okay with you.

Reply:
That'd be great.  Thanks.